internal/secrets/localsecrets/localsecrets.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package localsecrets provides a way to encrypt and decrypt small messages
+// without making network calls to a third party service.
+package localsecrets
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"golang.org/x/crypto/nacl/secretbox"
+	"io"
+)
+
+// secretKeeper holds a secret for use in symmetric encryption,
+// and implementations of driver.Encryper and driver.Decrypter.
+type secretKeeper struct {
+	secretKey [32]byte // secretbox key size
+	Encrypter driver.Encrypter
```

Comment:
**Blocking review question:** I could be missing something here, but it seems more direct to remove the field indirections and have the `Encrypt` and `Decrypt` methods below be methods of `secretKeeper`. Is there a reason to allow modification of these fields?

Reply:
summarizing in-person discussion: I am going to add a `Crypter` interface type to the driver, to reflect the "inherent paired-ness" of the encryption and decryption functionality ("any" Encrypter-Decrypter pair is not necessarily compatible, only those which are based on the same underlying credentials/secret), and refactor this package to remove the discrete `encrypter` and `decrypter` types and make `Encrypt` and `Decrypt` as methods directly on the exported type `SecretKeeper`, which will then be a `Crypter`.